compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/types/IrTypeUtils.kt

```diff
@@ -90,3 +96,23 @@ fun IrType.isNullable(): Boolean = DFS.ifAny(
         }
     }
 )
+
+val IrType.isBoxedArray: Boolean
+    get() = classOrNull?.owner?.fqNameWhenAvailable == KotlinBuiltIns.FQ_NAMES.array.toSafe()
+
+fun IrType.getArrayElementType(irBuiltIns: IrBuiltIns): IrType =
+    if (isBoxedArray)
+        ((this as IrSimpleType).arguments.single() as IrTypeProjection).type
+    else {
+        val classifier = this.classOrNull!!
+        irBuiltIns.primitiveArrayElementTypes[classifier]
+            ?: throw AssertionError("Primitive array expected: $classifier")
+    }
+
+fun IrType.toArrayType(irBuiltIns: IrBuiltIns): IrType =
```

Comment:
Suggest renaming to `toArrayOrPrimitiveArrayType`

Reply:
Renamed.